### ISSUE TITLE
User defined units not found with `magnetic_pressure`

### ISSUE THREAD

**Issue by maxbeegee** [yt-project/yt#2032]
### Bug report

**Bug summary**

yt does not seem to be able to use user defined units with the field `magnetic_pressure`

**Code for reproduction**

```python
ds = yt.load(some_file)
ds.define_unit("ptest", (1.0, "dyn/cm**2"))
ad = ds.all_data()
ad['pressure'].min().to('ptest') # note that this works
ad['magnetic_pressure'].min().to('ptest') # this fails
```

**Actual outcome**

```
Traceback (most recent call last):
  File "yt_but_magnetic_pressure.py", line 6, in <module>
    print(ad['magnetic_pressure'].min().to('ptest'))
  File "/home1/05311/maxbg/libs/python/my_yt/yt/yt/units/yt_array.py", line 639, in to
    return self.in_units(units, equivalence=equivalence, **kwargs)
  File "/home1/05311/maxbg/libs/python/my_yt/yt/yt/units/yt_array.py", line 621, in in_units
    new_units = _unit_repr_check_same(self.units, units)
  File "/home1/05311/maxbg/libs/python/my_yt/yt/yt/units/yt_array.py", line 269, in _unit_repr_check_same
    other_units = Unit(other_units, registry=my_units.registry)
  File "/home1/05311/maxbg/libs/python/my_yt/yt/yt/units/unit_object.py", line 268, in __new__
    unit_data = _get_unit_data_from_expr(unit_expr, registry.lut)
  File "/home1/05311/maxbg/libs/python/my_yt/yt/yt/units/unit_object.py", line 577, in _get_unit_data_from_expr
    return _lookup_unit_symbol(str(unit_expr), unit_symbol_lut)
  File "/home1/05311/maxbg/libs/python/my_yt/yt/yt/units/unit_object.py", line 668, in _lookup_unit_symbol
    "symbols." % symbol_str)
yt.units.unit_registry.UnitParseError: Could not find unit symbol 'ptest' in the provided symbols.
```

**Expected outcome**

The snippet should run and the two lines give the same result.

**Version Information**

  * Operating System: unix
  * Python Version: Python 2.7.14 :: Anaconda custom (64-bit)
  * yt version: 3.5.dev0
  * Other Libraries (if applicable): 

yt installed via from github.

**Comment by jzuhone** [comment]
I'll have a look at this one. @maxbeegee, do you have a particular test dataset this happens on?

**Comment by maxbeegee** [comment]
Thanks. This should do it: http://use.yt/upload/5f2d0a4e (~170MB)

**Comment by jzuhone** [comment]
This is happening because the magnetic field units are not getting the dataset's unit registry. Still not sure why, investigating.

**Comment by ngoldbaum** [comment]
Fixed by #2035
